src/main/java/io/cdap/plugin/gcp/common/ExceptionUtils.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.common;
+
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.HttpResponseException;
+import io.cdap.cdap.api.exception.ErrorCategory;
+import io.cdap.cdap.api.exception.ErrorCategory.ErrorCategoryEnum;
+import io.cdap.cdap.api.exception.ErrorUtils;
+import io.cdap.cdap.api.exception.ProgramFailureException;
+import java.io.IOException;
+
+/**
+ * Utility class to handle exceptions.
+ */
+public class ExceptionUtils {
+
+  /** Functional interfaces for lambda-friendly method invocations */
+  @FunctionalInterface
+  public interface IOOperation {
+    void execute() throws IOException;
+  }
+
+  /**
+   * Functional interfaces for lambda-friendly method invocations.
+   *
+   * @param <T> the return type of the function
+   */
+  @FunctionalInterface
+  public interface IOFunction<T> {
```

Comment:
nit: Consider using Callable / Runnable instead of Function / Operation. This terms are more used in Java